src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/IConnectionControl.cs

```diff
@@ -8,5 +8,8 @@ public interface IConnectionControl
         void Pause();
         void Resume();
         void End(ProduceEndType endType);
+
+        void NotifyRequestStarted();
```

Comment:
What do you suggest? To me it makes sense to expose these here - you use these to notify the signal the connection about request processing events so it can react accordingly.

Do you think I should introduce a new interface? `IRequestStatusNotification` or something along those lines?